src/highrl/utils/teacher_utils.py

```diff
@@ -170,3 +173,12 @@ def get_reward(
     )
 
     return reward + too_close_to_goal_penality
+
+
+def load_env_generator(model_type: str, cfg):
+    if model_type == "RL":
+        model = PPO.load(cfg.generator_path)
+        return model
+
+    elif model_type == "SL":
+        model = th.load(cfg.generator_path)
```

Comment:
You forgot to return the model, if you're loading a `SL` model. Also, there is no need for the `elif` statement here. 
```python
def load_env_generator(model_type: str, cfg):
    if model_type == "RL":
        return PPO.load(cfg.generator_path)
    return th.load(cfg.generator_path)
```